unison-share-api/src/Unison/Server/Endpoints/DefinitionSummary.hs

```diff
@@ -108,3 +111,55 @@ serveTermSummary codebase exactNameSH@(ExactName {name = termName, ref = shortHa
       if Reference.isBuiltin reference
         then BuiltinObject termSig
         else UserObject termSig
+
+type TypeSummaryAPI =
+  "definitions" :> "types" :> "qualified" :> Capture "fqn" (ExactName Name SH.ShortHash) :> "summary"
+    :> QueryParam "rootBranch" ShortBranchHash
+    :> QueryParam "relativeTo" Path.Path
+    :> QueryParam "renderWidth" Width
+    :> APIGet TypeSummary
+
+instance ToSample TypeSummary where
+  toSamples _ = noSamples
+
+data TypeSummary = TypeSummary
+  { fqn :: Name,
+    hash :: SH.ShortHash,
+    summary :: DisplayObject SyntaxText SyntaxText,
+    tag :: TypeTag
+  }
+  deriving (Generic, Show)
+
+instance ToJSON TypeSummary where
+  toEncoding = genericToEncoding defaultOptions
+
+deriving instance ToSchema TypeSummary
+
+serveTypeSummary ::
+  Codebase IO Symbol Ann ->
+  ExactName Name SH.ShortHash ->
+  Maybe ShortBranchHash ->
+  Maybe Path.Path ->
+  Maybe Width ->
+  Backend IO TypeSummary
+serveTypeSummary codebase exactNameSH@(ExactName {name, ref = shortHash}) _mayRoot _relativeTo mayWidth = do
```

Comment:
We actually don't end up caring about the root or relativeTo for now, but we may need them in the future, especially if we end up doing access checks based on path, so I'll leave them in here for symmetry with term summaries.